test/src/org/apache/jmeter/report/processor/StatisticsSummaryDataTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.apache.jmeter.report.processor;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.HdrHistogram.Histogram;
+import org.LatencyUtils.LatencyStats;
+import org.LatencyUtils.SimplePauseDetector;
+import org.junit.Before;
+import org.junit.Test;
+
+public class StatisticsSummaryDataTest {
+    private StatisticsSummaryData statisticsSummaryData;
+    private SimplePauseDetector defaultPauseDetector = new SimplePauseDetector();
+    private static long lowestTrackableLatency = 1000000L;
+    private static long highestTrackableLatency = 3600000000000L;
+    private static int numberOfSignificantValueDigits = 2;
+    private static int intervalEstimatorWindowLength = 1024;
+    private static long intervalEstimatorTimeCap = 10000000000L;
+    @Before
+    public void setUp() {
+        statisticsSummaryData = new StatisticsSummaryData();
+    }
+
+    @Test
+    public void testStatisticsSummaryData() {
+        LatencyStats latencyStats = new LatencyStats(lowestTrackableLatency, highestTrackableLatency,
+                numberOfSignificantValueDigits, intervalEstimatorWindowLength, intervalEstimatorTimeCap,
+                defaultPauseDetector);
+        Histogram histogram = new Histogram(latencyStats.getIntervalHistogram());
+        long[] values = new long[] { 10L, 9L, 5L, 6L, 3L, 8L, 2L, 7L, 4L };
+        for (long l : values) {
+            statisticsSummaryData.addValue(l);
+            statisticsSummaryData.setMax(l);
+            statisticsSummaryData.setMin(l);
+            statisticsSummaryData.incTotal();
+            histogram.recordValue(l*1000000L);
+        }
+        // test max/min
+        assertTrue(10 == statisticsSummaryData.getMax());
```

Comment:
Please avoid writing tests like that.
It would be hard to analyze especially at Travis/Jenkins CI.

Please use something like `assertEquals(Arrays.toString(values) + ".getMax()", 10, statisticsSummaryData.getMax())`

General rule: `assertTrue` and `assertFalse` are almost always bad.